pkg/apis/core/v1beta1/constants/types_constants.go

```diff
@@ -730,6 +730,8 @@ const (
 	LabelWorkerPoolDeprecated = "worker.garden.sapcloud.io/group"
 	// LabelWorkerPoolSystemComponents is a constant that indicates whether the worker pool should host system components
 	LabelWorkerPoolSystemComponents = "worker.gardener.cloud/system-components"
+	// LabelWorkerPoolOperatingSystemConfig is the name of the applied OperatingSystemConfig secret
+	LabelWorkerPoolOperatingSystemConfig = "worker.gardener.cloud/operating-system-config"
```

Comment:
```suggestion
	LabelWorkerPoolOperatingSystemConfigSecretName = "worker.gardener.cloud/operating-system-config-secret-name"
```

WDYT? This would make it more explicit. Or, perhaps even better

```suggestion
	LabelWorkerPoolGardenerNodeAgentSecretName = "worker.gardener.cloud/gardener-node-agent-secret-name"
```

The fact, that this secret contains a `OperatingSystemConfig` is a detail that should not be encoded into the label name, don't you think?

Reply:
I've changed the name to `LabelWorkerPoolGardenerNodeAgentSecretName`. Although, my impression is that "operating-system-config" would be generic enough for anyone not familiar with Gardener to interpret it as a general name, not a specify gardener component. But I'm fine with either variant.